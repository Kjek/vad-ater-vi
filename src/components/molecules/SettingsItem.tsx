import TextBox from '@component/admin/TextBox';
import Text from '@component/atoms/Text';
import type { ChangeEventHandler } from 'react';

interface SettingsItemProps {
  title: string;
  inputDefaultValue: string;
  inputPlaceholder: string;
  onChange: ChangeEventHandler<HTMLInputElement>;
}

const SettingsItem = ({
  title,
  inputDefaultValue,
  inputPlaceholder,
  onChange,
}: SettingsItemProps) => {
  return (
    <div className='mb-4 flex items-center gap-4'>
      <Text variant='h4' className='w-2/5 text-sm'>
        {title}
      </Text>
      <TextBox
        id={title.toDotNotation()}
        defaultValue={inputDefaultValue}
        placeholder={inputPlaceholder}
        onChange={onChange}
      />
    </div>
  );
};

export default SettingsItem;
