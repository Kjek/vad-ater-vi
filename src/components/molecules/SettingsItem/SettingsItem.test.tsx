import { fireEvent, render } from '@testing-library/react';
import { describe, expect, it, vi } from 'vitest';
import SettingsItem from './SettingsItem';

String.prototype.toDotNotation = vi.fn();

describe(SettingsItem, () => {
  it('renders correctly', () => {
    const onChange = vi.fn();
    const { container } = render(
      <SettingsItem
        title='Example'
        inputDefaultValue={''}
        inputPlaceholder='Type something here...'
        onChange={onChange}
      />
    );
    expect(container).toBeInTheDocument();
    expect(container).toBeVisible();
  });

  it('displays value correctly', () => {
    const onChange = vi.fn();
    const { getByDisplayValue } = render(
      <SettingsItem
        title='Example'
        inputDefaultValue={'Default Value'}
        inputPlaceholder='Type something here...'
        onChange={onChange}
      />
    );

    const textBox = getByDisplayValue('Default Value');
    fireEvent.change(textBox, { target: { value: 'test' } });
    expect(onChange).toHaveBeenCalledOnce();
    expect(textBox).toHaveValue('test');
  });

  it('matches snapshot', () => {
    const defaultRender = render(
      <SettingsItem
        title='Example'
        inputDefaultValue={''}
        inputPlaceholder='Type something here...'
        onChange={vi.fn()}
      />
    );
    expect(defaultRender).toMatchSnapshot();
  });
});
