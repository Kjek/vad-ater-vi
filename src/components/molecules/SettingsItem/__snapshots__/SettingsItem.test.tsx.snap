// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SettingsItem > matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mb-4 flex items-center gap-4"
      >
        <h4
          class="font-bold text-gray-800 dark:text-gray-300 w-2/5 text-sm"
        >
          Example
        </h4>
        <input
          class="w-full self-center rounded-lg border border-gray-300 bg-gray-50 p-2.5 text-sm text-gray-800 outline-none focus:border-gray-300 focus:ring-gray-300 dark:border-gray-600 dark:bg-gray-800 dark:text-gray-300 dark:placeholder-gray-400 dark:focus:border-gray-600 dark:focus:ring-gray-600"
          placeholder="Type something here..."
          type="text"
          value=""
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mb-4 flex items-center gap-4"
    >
      <h4
        class="font-bold text-gray-800 dark:text-gray-300 w-2/5 text-sm"
      >
        Example
      </h4>
      <input
        class="w-full self-center rounded-lg border border-gray-300 bg-gray-50 p-2.5 text-sm text-gray-800 outline-none focus:border-gray-300 focus:ring-gray-300 dark:border-gray-600 dark:bg-gray-800 dark:text-gray-300 dark:placeholder-gray-400 dark:focus:border-gray-600 dark:focus:ring-gray-600"
        placeholder="Type something here..."
        type="text"
        value=""
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
